crates/holochain/src/core/ribosome/host_fn/accept_countersigning_preflight_request.rs

```diff
@@ -139,6 +139,229 @@ pub mod wasm_test {
+        let (countersigned_action_hash_alice, countersigned_entry_hash_alice): (ActionHash, EntryHash) = conductor
+            .call(
+                &alice,
+                "create_a_countersigned_thing_with_entry_hash",
+                vec![alice_response.clone(), bob_response.clone()],
+            )
+            .await;
+
+        // Bob tries to do the same thing but after timeout.
+        tokio::time::sleep(std::time::Duration::from_millis(2000)).await;
+        let bob_result: Result<ActionHash, _> = conductor
+        .call_fallible(
+            &bob,
+            "create_a_countersigned_thing",
+            vec![alice_response.clone(), bob_response.clone()],
+        )
+        .await;
+        expect_chain_lock_expired(bob_result);
+
+        // At this point Alice's session entry is a liability so can't exist.
+        let alice_agent_activity_alice_observed_after: AgentActivity = conductor
+            .call(
+                &alice,
+                "get_agent_activity",
+                GetAgentActivityInput {
+                    agent_pubkey: alice_pubkey.clone(),
+                    chain_query_filter: ChainQueryFilter::new(),
+                    activity_request: ActivityRequest::Full,
+                }
+            ).await;
+        let alice_agent_activity_bob_observed_after: AgentActivity = conductor
+            .call(
+                &bob,
+                "get_agent_activity",
+                GetAgentActivityInput {
+                    agent_pubkey: alice_pubkey.clone(),
+                    chain_query_filter: ChainQueryFilter::new(),
+                    activity_request: ActivityRequest::Full,
+                }
+            ).await;
+        let bob_agent_activity_alice_observed_after: AgentActivity = conductor
+            .call(
+                &alice,
+                "get_agent_activity",
+                GetAgentActivityInput {
+                    agent_pubkey: bob_pubkey.clone(),
+                    chain_query_filter: ChainQueryFilter::new(),
+                    activity_request: ActivityRequest::Full,
+                }
+            ).await;
+        let bob_agent_activity_bob_observed_after: AgentActivity = conductor
+            .call(
+                &bob,
+                "get_agent_activity",
+                GetAgentActivityInput {
+                    agent_pubkey: bob_pubkey.clone(),
+                    chain_query_filter: ChainQueryFilter::new(),
+                    activity_request: ActivityRequest::Full,
+                }
+            ).await;
+
+        assert_eq!(alice_agent_activity_alice_observed_before, alice_agent_activity_alice_observed_after);
+        assert_eq!(alice_agent_activity_bob_observed_before, alice_agent_activity_bob_observed_after);
+        assert_eq!(bob_agent_activity_alice_observed_before, bob_agent_activity_alice_observed_after);
+        assert_eq!(bob_agent_activity_bob_observed_before, bob_agent_activity_bob_observed_after);
```

Comment:
Do you want to make any assertion about what the observations actually are, or just that they're equal?

Reply:
nah i don't think it matters for this test, sameness is good enough